src/components/TodoModal/TodoModal.tsx

```diff
@@ -2,11 +1,47 @@
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+import { Todo } from '../../types/Todo';
+
+interface TodoModalProps {
+  onClose: (value: undefined) => void;
+  post: Todo;
+}
+
+export const TodoModal: React.FC<TodoModalProps> = ({
+  onClose,
+  post: { userId, id, title, completed },
+}) => {
+  const [user, setUser] = useState<User>();
+  const [loaded, setLoaded] = useState(false);
+
+  let name, email;
+
+  const onEscapeEvent = (event: KeyboardEvent) => {
+    if (event.key === 'Escape') {
+      onClose(undefined);
+    }
+  };
+
+  if (user) {
+    ({ name, email } = user);
+  }
+
+  useEffect(() => {
+    getUser(userId).then(result => {
+      setUser(result);
+      setLoaded(true);
+    });
+
+    document.addEventListener('keydown', onEscapeEvent);
```

Comment:
Consider removing the event listener for the Escape key when the component unmounts to prevent memory leaks. You can do this by returning a cleanup function from the `useEffect` hook.